v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/utils/ShardIdFetcherImpl.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates.utils;
+
+import com.google.cloud.teleport.v2.spanner.migrations.schema.Schema;
+import com.google.cloud.teleport.v2.spanner.migrations.schema.SpannerTable;
+import com.google.cloud.teleport.v2.spanner.utils.IShardIdFetcher;
+import com.google.cloud.teleport.v2.spanner.utils.ShardIdRequest;
+import com.google.cloud.teleport.v2.spanner.utils.ShardIdResponse;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Default implementation to get the shard identifier. */
+public class ShardIdFetcherImpl implements IShardIdFetcher {
```

Comment:
nit: The class can be renamed to reflect that this is Default ShardIdFetcher based on shardId column in session file.

Reply:
Leaving it for now, since going forward session file itself will be optional, then this will not be the Default. Also am not sure if in future the file will still be named as session file or schema mapping file, will re-consider this when that item is picked up.